Tests/SwiftParserTest/TypeTests.swift

```diff
@@ -15,16 +15,30 @@
 import XCTest
 
 final class TypeTests: XCTestCase {
-
   func testMissingColonInType() {
-    assertParse(
-      """
-      var foo 1️⃣Bar = 1
-      """,
-      diagnostics: [
-        DiagnosticSpec(message: "expected ':' in type annotation")
-      ]
-    )
+    let testCases: [UInt: (fixIt: String, fixedSource: String)] = [
+      #line: ("join the identifiers together", "var fooBar = 1"),
+      #line: ("insert ':'", "var foo: Bar = 1"),
+    ]
+
+    for (line, testCase) in testCases {
+      assertParse(
+        """
+        var foo 1️⃣Bar = 1
+        """,
+        diagnostics: [
+          DiagnosticSpec(message: "expected ':' in type annotation", fixIts: ["insert ':'"], line: line),
+          DiagnosticSpec(
+            message: "found an unexpected second identifier in variable; is there an accidental break?",
+            fixIts: ["join the identifiers together"],
+            line: line
+          ),
```

Comment:
I just wouldn’t bother generating this diagnostic. I think `expected ':' in type annotation` is a perfectly adequate diagnostic for the test case. IIUC that would allow you to remove most of the code you wrote in ParseDiagnsoticGenerator.swift

Reply:
I'm not sure I understand this. If I remove the code in `ParseDiagnsoticGenerator.swift`, it will fail as we are not generating the expected diagnostics.